### ISSUE TITLE
Remove JENKINS_EXPLICIT_VERSION

### ISSUE THREAD

**Issue by ihmccreery** [kubernetes/kubernetes#17742]
As far as I can tell, we should never be using `JENKINS_EXPLICIT_VERSION`, instead using `JENKINS_PUBLISHED_VERSION`, e.g. `ci/latest-1.1`.  `JENKINS_EXPLICIT_VERSION` is not actionable; changing something won't fix tests pinned to a particular version.

`JENKINS_EXPLICIT_VERSION` would be useful for release validation, but we're not set up for doing that right now, and I don't think we will be any time soon.  That will require an explicit effort that I don't think we're ready to take on.

cc @wonderfly @kubernetes/goog-testing

**Comment by ihmccreery** [comment]
Actually, even for release validation, we should probably be floating on `release/latest` or `release/latest-1`.

**Comment by wonderfly** [comment]
> Actually, even for release validation, we should probably be floating on release/latest or release/latest-1.

The way we are using ```JENKINS_EXPLICIT_VERSION``` is exceptional. We are not qualifying any version of k8s or GKE, we are qualifying an OS image. The Jenkins jobs that use this flag are all CIs for Trusty images. For that, we want to pin to a _fixed_ version of k8s, because otherwise we won't be able to know if it's the updated image or the upgraded k8s is at fault for a test failure. ```ci/latest-1.1``` is _not fixed_ as far as I know - it will change to 1.1.2 _silently_ once 1.1.2 is out. We don't want that kind of surprise in our image qualification tests, yet.

> JENKINS_EXPLICIT_VERSION is not actionable; changing something won't fix tests pinned to a particular version.

That's due to the fact that tests are k8s binaries are synchronously versioned. When we pin to a fixed version, we _do_ like the test fixes in a later version, but we don't want the upgraded k8s in that later version.

+cc @andyzheng0831 @roberthbailey for more context.

**Comment by ihmccreery** [comment]
> We don't want that kind of surprise in our image qualification tests, yet.

Wouldn't you rather have an early signal that your image won't work with the latest version of Kubernetes?  (I'm totally understanding if the answer is 'no', just want to make sure we cover our bases.)

Somewhat related to #17734.

**Comment by andyzheng0831** [comment]
That is something we will have for the next step. Currently, the trusty jenkins jobs are mainly for qualification of our image, to catch breakage caused by image itself as early as possible. That is why we need to pin to a known-clean-latest-k8s-release. Next, we will also have jobs pined to k8s head for k8s qualification. If we pin to k8s head and image change head, when something is wrong, it is very hard to debug. So, decoupling the k8s and image combinations is helpful for us

**Comment by wonderfly** [comment]
> Wouldn't you rather have an early signal that your image won't work with the latest version of Kubernetes?

We would. But as @andyzheng0831 said, it's our next step. And even when that comes along, we will still need this set of jobs just in case we aren't able to get a green Jenkins build with the latest k8s, but still have to make an image release cut (in that case we'll release the image with an earlier k8s version that we have fully tested in parallel). Does this make sense? Otherwise, we have an internal design doc that details the plan. Ping me for the link if you are interested.

**Comment by ihmccreery** [comment]
Ack.  Thanks for the clarification.  Closing.
